recipes/libavrocpp/1.10.1/conanfile.py

```diff
@@ -0,0 +1,100 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class LibavrocppConan(ConanFile):
+    name = "libavrocpp"
+    version = "1.10.1"
+    license = "Apache License 2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Avro is a data serialization system."
+    homepage = "https://avro.apache.org/"
+    topics = ("serialization", "deserialization")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_snappy": [True, False],
+    }
+    default_options = {"shared": False, "fPIC": True, "with_snappy": True}
+    generators = "cmake", "cmake_find_package"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        compiler = str(self.settings.compiler)
+        compiler_version = Version(self.settings.compiler.version.value)
+
+        minimal_version = {
+            "Visual Studio": "10",
+            "gcc": "4.9.1",
+            "clang": "3.3",
+            "apple-clang": "5",
+        }
+
+        if compiler in minimal_version and compiler_version < minimal_version[compiler]:
```

Comment:
We like to print a warning when the the compiler is not known.

conan-io/conan#8002 has a nice boilerplate you can copy...

If you require the granularity for the compiler version you need to use the `lazy_semver_lt` from Space https://github.com/conan-io/conan-center-index/commit/325c293a35ec94477c9e5197190190d2d00add97

None of the compilers excluded are support on CCI this is not required and can be removed.